src/callbacks.c

```diff
@@ -1,4 +1,12 @@
 #include "curl-common.h"
+#include "utils.h"
+
+int default_callback_progress(void *dummy,
+                              double dltotal, double dlnow,
+                              double ultotal, double ulnow)
+{
+  return !(R_ToplevelExec(check_interrupt_fn, NULL));
```

Comment:
Crude, but looks OK at least on Linux. Could also run under valgrind to better check for leaks (nothing apparent though)...

```
#!/bin/bash

R --no-save -e "library(curl); h = new_handle(); replicate(100, tryCatch(curl_fetch_memory('http://httpbin.org/delay/10', h), error=function(e) cat('Cazart!\n', file=stderr())))" &
pid=$!
sleep 5

j=1
while test $j -lt 100;do
  echo "Interrupt number $j"
  sleep 1
  kill -2 $!
  j=$(($j + 1))
done

wait
```